### ISSUE TITLE
Group layouts not working in turbopack 

### ISSUE THREAD

**Issue by joacub** [vercel/next.js#52621]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Last canary version 13.4.10-6
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App Router, Turbopack (--turbo)

### Link to the code that reproduces this issue or a replay of the bug

No response 

### To Reproduce

Having this route segments will break in the home route and all routes inside group main

[lang]/(main)/page.tsx
layout.tsx

when accesing routes in main group all routes fails as there is no matching routes

In webpack works fine.

### Describe the Bug

No response 

### Expected Behavior

No response 

<sub>[WEB-1298](https://linear.app/vercel/issue/WEB-1298/group-layouts-not-working-in-turbopack)</sub>

**Comment by balazsorban44** [comment]
Hi @joacub, I've gone through your [previously reported issues](https://github.com/vercel/next.js/issues?q=is%3Aissue+author%3Ajoacub+label%3A%22please+add+a+complete+reproduction%22+), and almost all lacked a reproduction.

Please note that the Next.js issue tracker is for reporting reproducible issues, and thus they need to follow the issue template. Not providing a proper reproduction makes it hard/impossible to investigate.

I kindly ask you to not open more issues without following the required steps, as described in the issue template. Not following the template can result in issues being converted to [discussions](https://github.com/vercel/next.js/discussions). If you are not certain if something is a Next.js bug, you can use Discussions, or our [Discord server](https://discord.com/invite/bUG2bvbtHy) where community members might be able to help with debugging your project further. 

Thanks for understanding! :pray:

**Comment by balazsorban44** [comment]
> is the explanation of the issue not enough to understand the issue and check

There is a high volume of issues coming in, the ones that come with a reproduction help us help you the fastest.

Please read https://github.com/vercel/next.js/issues/52621#issuecomment-1633970602 for a full explanation. We absolutely don't want to discourage you from reporting valid bugs, but we rely on you and the rest of our community to cooperate for fast resolutions for everyone's success, that's why we've put effort into creating the issue template, which we expect everyone to respect and fill out accordingly.

**Comment by joacub** [comment]
> > is the explanation of the issue not enough to understand the issue and check
> 
> There is a high volume of issues coming in, the ones that come with a reproduction help us help you the fastest.
> 
> Please read [#52621 (comment)](https://github.com/vercel/next.js/issues/52621#issuecomment-1633970602) for a full explanation. We absolutely don't want to discourage you from reporting valid bugs, but we rely on you and the rest of our community to cooperate for fast resolutions for everyone's success, that's why we've put effort into creating the issue template, which we expect everyone to respect and fill out accordingly.

I completely understand your perspective and your request. Please don't misunderstand me, but I regret to inform you that I am unable to fulfill it. Due to time constraints, I am unable to engage in reproductions. Additionally, I believe that reproductions are often unnecessary as they tend to be simple in nature and can be easily resolved without the need for reproduction.

But yeah i totally understand have a good one

**Comment by joacub** [comment]
this may help: 

```shell
Non matching path (es) provided for /[lang]/(main)

Debug info:
- Execution of get_from_source failed
- Execution of resolve_source_request failed
- Execution of NodeRenderContentSource::get failed
- Non matching path (es) provided for /[lang]/(main)
```

**Comment by kmvan** [comment]
Same issue. Watching.

**Comment by zfm-andreasmatuszewski** [comment]
Any updates here?

**Comment by RafaelZasas** [comment]
I'm having the same issue
Any update?

**Comment by padmaia** [comment]
This appears to be fixed with v13.5+
